framework/validators/CompareValidator.php

```diff
@@ -85,28 +85,28 @@ public function init()
         if ($this->message === null) {
             switch ($this->operator) {
                 case '==':
-                    $this->message = Yii::t('yii', '{attribute} must be repeated exactly.');
+                    $this->message = Yii::t('yii', '{attribute} must be equal to "{compareInternal}".');
```

Comment:
> Also, when you use compare validator to ensure password is repeated the same, this new message doesn't read very naturally and will almost always be changed.

We have a problem here: when using `compareValue` would not make sense a 'must be repeated exactly' message. Instead of this, we would like to see a 'must be equal to {compareValueOrAttribute}'.

We cant use a condition to check `if ($this->compareValue === null)` in `init()` because in this method this condition would be ever true.

I just can see two options: 

1. Making two messages in a array for the == or === operators and choose the right later, checking if the `compareValue` is null or not.
2. Using an unique message for both cases (and `repeated exactly` would not fit very well anymore).